cadl-tests/src/main/java/com/cadl/partialupdate/PartialUpdateClientBuilder.java

```diff
@@ -0,0 +1,255 @@
+        Configuration buildConfiguration =
+                (configuration == null) ? Configuration.getGlobalConfiguration() : configuration;
+        if (httpLogOptions == null) {
+            httpLogOptions = new HttpLogOptions();
+        }
+        if (clientOptions == null) {
+            clientOptions = new ClientOptions();
+        }
+        List<HttpPipelinePolicy> policies = new ArrayList<>();
+        String clientName = properties.getOrDefault(SDK_NAME, "UnknownName");
+        String clientVersion = properties.getOrDefault(SDK_VERSION, "UnknownVersion");
+        String applicationId = CoreUtils.getApplicationId(clientOptions, httpLogOptions);
+        policies.add(new UserAgentPolicy(applicationId, clientName, clientVersion, buildConfiguration));
+        policies.add(new RequestIdPolicy());
+        policies.add(new AddHeadersFromContextPolicy());
+        HttpHeaders headers = new HttpHeaders();
+        clientOptions.getHeaders().forEach(header -> headers.set(header.getName(), header.getValue()));
+        if (headers.getSize() > 0) {
+            policies.add(new AddHeadersPolicy(headers));
+        }
+        policies.addAll(
+                this.pipelinePolicies.stream()
+                        .filter(p -> p.getPipelinePosition() == HttpPipelinePosition.PER_CALL)
+                        .collect(Collectors.toList()));
+        HttpPolicyProviders.addBeforeRetryPolicies(policies);
+        policies.add(ClientBuilderUtil.validateAndGetRetryPolicy(retryPolicy, retryOptions, new RetryPolicy()));
+        policies.add(new AddDatePolicy());
+        policies.add(new CookiePolicy());
+        policies.addAll(
+                this.pipelinePolicies.stream()
+                        .filter(p -> p.getPipelinePosition() == HttpPipelinePosition.PER_RETRY)
+                        .collect(Collectors.toList()));
+        HttpPolicyProviders.addAfterRetryPolicies(policies);
+        policies.add(new HttpLoggingPolicy(httpLogOptions));
+        HttpPipeline httpPipeline =
+                new HttpPipelineBuilder()
+                        .policies(policies.toArray(new HttpPipelinePolicy[0]))
+                        .httpClient(httpClient)
+                        .clientOptions(clientOptions)
+                        .build();
+        return httpPipeline;
+    }
+
+    /**
+     * Builds an instance of PartialUpdateAsyncClient class.
+     *
+     * @return an instance of PartialUpdateAsyncClient.
+     */
+    @Generated
+    public PartialUpdateAsyncClient buildAsyncClient() {
+        return new PartialUpdateAsyncClient(buildInnerClient().getPartialUpdateOps());
+    }
+
+    /**
+     * Builds an instance of PartialUpdateClient class.
+     *
+     * @return an instance of PartialUpdateClient.
+     */
+    @Generated
+    public PartialUpdateClient buildClient() {
+        return new PartialUpdateClient(new PartialUpdateAsyncClient(buildInnerClient().getPartialUpdateOps()));
+    }
+
+    public void test() {}
```

Comment:
manual code here. I did not add complex cases, because they are covered in partial update unit tests previously.

Reply:
any one in client?